api/db.py

```diff
@@ -119,6 +119,37 @@ def fetchAllPreferredEditions(
             editionsQuery.offset(offset).limit(perPage).all(),
         )
 
+    def fetchEditions(self, editionIDs):
+        """Fetch the editions in the given list, respecting the order
+        of the passed in list
+        """
+
+        # First build a CTE of the passed in ids and their index in the
+        # given list
+        editionIdCTE = sa.select(
+            sa.values(
+                sa.column("idx", sa.Integer),
+                sa.column("edition_id", sa.Integer),
+                name="subquery",
+            ).data(list(enumerate(editionIDs)))
+        ).cte("edition_ids")
```

Comment:
I like your idea of using a CTE for building this query. I've never had experience using CTEs before, so your use of it is very interesting as a way to create more complex queries. With your use of it here, I might start using them for some queries I create in the future.